luigi/contrib/ftp.py

```diff
@@ -127,10 +127,14 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
+        path_parts = path.split('/')
+        path = '/'.join(path_parts[:-1])
+        fn = path_parts[-1]
```

Comment:
I'm not sure I understand you. There is no loop within this method (at least not directly). Line 137 is checking to see if the filename exists in the list of files which was the result of `nlist(path)`.